### PR TITLE
[client] build out form interface and support saving + submitting

### PR DESCRIPTION
**Pull request by WilhelmWillie**
Screen Shot 2019-10-31 at 12 02 09 AM

* Build out the form interface for the profile step
* Integrate with the hacker profile API
* Allow saving (users can save data and come back later)
* Allow submission (adjusts users status and enables them to fill out the application)

TO-DO for future PR:
* Better input validation
* Code cleanup
* Get resume uploading set-up

### CODE REVIEW

**Comment by saresend** [comment]
Do you want a review on this?

**Comment by WilhelmWillie** [comment]
ooh give me a bit, gonna get it to hit the PUT /api/profile route and do some stuff

File: api/database/migrations/20191031003446-profile-additional-fields.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"use strict";
+
+module.exports = {
```
Comment (WilhelmWillie) [inline api/database/migrations/20191031003446-profile-additional-fields.js]: Added some new fields.. also modified existing fields to make sure we were consistent with `enum` values between client and server

File: api/models/hackerProfile.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ module.exports = (sequelize, DataTypes) => {
     {
       gender: {
         type: DataTypes.ENUM,
-        values: ["male", "female", "other"]
+        values: ["male", "female", "non-binary", "other", "no-say"]
```
Comment (WilhelmWillie) [inline api/models/hackerProfile.js]: added more specific gender options

File: assets/chevron.svg [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (WilhelmWillie) [inline assets/chevron.svg]: lmao @ prettier

File: lib/formSubmission.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  const response = await fetch("/api/profile", {
+    method: "PUT",
+    body: JSON.stringify(profile),
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+
+  const data = await response.json();
+  setSaved(true);
```
Comment (WilhelmWillie) [inline lib/formSubmission.tsx]: TODO: Error handling and checking

File: api/database/migrations/20191031003446-profile-additional-fields.js (hunk 1 above)
Comment (saresend) [reply api/database/migrations/20191031003446-profile-additional-fields.js]: I support this

File: api/hackerProfile.js [hunk 5]
```diff
@@ -23,13 +23,102 @@ router.get("/", async (req, res) => {
 });
 
 router.put("/", async (req, res) => {
-  // TODO: Do input validation
-  const newHackerProfile = await models.HackerProfile.update(req.body, {
+  // Get the users current hacker profile
+  const currentHackerProfile = await models.HackerProfile.findOne({
+    where: { userId: req.user.id }
+  });
```
Comment (saresend) [inline api/hackerProfile.js]: How are you setting up the frontend, is this gonna always exist?

Comment (WilhelmWillie) [reply api/hackerProfile.js]: it should -- before they can access the profile form, a `GET /api/profile` request is made which should create a hacker profile if it doesn't exist

Comment (saresend) [reply api/hackerProfile.js]: kk sounds good

**Pull request merged**